File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {

        int l = 0;
        int r = numbers.length - 1;

        while ( l < r ) {

            int curSum = numbers[l] + numbers[r];

            if ( curSum > target ) {
                r--;
            }

            if ( curSum < target ) {
                l++;
            }

            if ( curSum == target ) {
                return new int[] {l + 1, r + 1};
            }
        }
        return new int[0];
    }
}


// public class Solution {
//     public int[] twoSum(int[] numbers, int target) {
//         for (int i = 0; i < numbers.length; i++) {
//             for (int j = i + 1; j < numbers.length; j++) {
//                 if (numbers[i] + numbers[j] == target) {
//                     return new int[] { i + 1, j + 1 };
//                 }
//             }
//         }
//         return new int[0];
//     }
// }

File: Data Structures & Algorithms/target-sum/submission-0.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        
        dp = {}

        def dfs(i,curSum):

            if (i,curSum) in dp:
                return dp[(i,curSum)]

            if i == len(nums):
                return 1 if curSum == target else 0

            plus  = dfs(i+1,curSum + nums[i])
            minus = dfs(i+1,curSum - nums[i])
            dp[(i,curSum)] = plus + minus
            return dp[(i,curSum)]

        return dfs(0,0)